### ISSUE TITLE
What should we do about config files?

### ISSUE THREAD

**Issue by ZPBerg** [PurdueCAM2Project/Embedded2#25]
I can think of these config files that are currently used by code in our repo:

src/jetson/config.json - arguments for main.py 
src/db/login.json - login information for our MySQL server and the HELPS machine (for image transfer)
scripts/params.json - hyperparameters to use when training the classifier in goggle_classifier

I know other repos would usually have default/suggested values in these kinds of files so that people who fork the repo know what information they need without having to look at the code itself. Should we do the same thing, given that this repo will be in stasis for a bit?

I'm concerned that the files would change any time someone pushes new code, which would be tedious each time we update our forks and have to re-update our config files. Also, usernames and passwords (login.json) obviously shouldn't appear in the repo ever, but it's possible that someone accidentally includes their credentials in a commit/PR, especially if they're new to Github.

**Comment by stephend017** [comment]
Instead of committing config files we should add them to the gitignore and document how to you can make them for local usage.

**Comment by jbagnara** [comment]
> Instead of committing config files we should add them to the gitignore and document how to you can make them for local usage.

Would it be possible to include skeleton config files in our repo, and add them to .gitignore to untrack further changes? Ideally anyone who uses our repo does not have to create their own .json files

**Comment by stephend017** [comment]
yes we can do that

**Comment by stephend017** [comment]
we should also look into creating a centralized location for these files as well

**Comment by pawar4** [comment]
Are you saying we should have a config folder for all our config files?

**Comment by stephend017** [comment]
yes I think that would be a good idea since we have several different config files.

**Comment by ZPBerg** [comment]
But login.json and params.json are for running main.py; params.json is for training a classifier (happens separately from main.py). 
If anything, in the future more hyperparameter files would appear for training other classifiers/detectors.

I don't think it makes sense to just put them all under the same "config" umbrella.

**Comment by stephend017** [comment]
we can make a sub folder for each script that contains the script and its configs. but everything for main.py should be in a central location
